patch.go

```diff
@@ -389,17 +389,13 @@ func (d *partialArray) add(key string, val *lazyNode) error {
 
 	cur := *d
 
-	if idx < 0 {
-		idx *= -1
-
-		if idx > len(ary) {
-			return fmt.Errorf("Unable to access invalid index: %d", idx)
-		}
-		idx = len(ary) - idx
-	}
-	if idx < 0 || idx >= len(ary) || idx > len(cur) {
```

Comment:
I believe the > and >= checks here are redundant so I removed one of them.

Reply:
Also, multiplying the index by -1 before reporting an error made the error message a bit misleading, so I check the index before altering it.